### PR TITLE
[FIX] Check presence of QR IBAN before checking validity

### PR DESCRIPTION
**Pull request by vrenaville**
* Purpose
We need to first check if the QR iban is filed before checking validity

https://github.com/OCA/l10n-switzerland/pull/584

### CODE REVIEW

**Review by ajaniszewska-dev — APPROVED** [review]

File: l10n_ch_qr_bill/models/res_bank.py [hunk 1]
```diff
@@ -44,14 +44,15 @@ def _validate_qr_iban(self, qr_iban):
...
                 "Not a valid Switzerland or Liechtenstein QR-IBAN."))
-        # Check first if it's a valid IBAN.
-        validate_iban(qr_iban)
-        # We sanitize first so that _check_qr_iban_range()
-        # can extract correct IID from IBAN to validate it.
-        sanitized_qr_iban = sanitize_account_number(qr_iban)
-        # Now, check if it's valid QR-IBAN (based on its IID).
-        if not self._check_qr_iban_range(sanitized_qr_iban):
-            raise ValidationError(_("QR-IBAN '%s' is invalid.") % qr_iban)
+        if qr_iban:
```
Comment (carlosserratoro) [inline l10n_ch_qr_bill/models/res_bank.py]: You already check in the previous ```if``` if ```qr_iban``` has a value.

```suggestion
```

**Review by carlosserratoro — APPROVED** [review]
If possible, remove the new ```if``` while keeping the indentation, that way you save a redundant check in every call.

**Comment by yvaucher** [comment]
@vrenaville I counter proposed an other fix to have `_validate...` function checking emptiness and making sure the constraint doesn't call validation for nothing.

@carlosserratoro Does it take take of your remark?

**Comment by carlosserratoro** [comment]
> @carlosserratoro Does it take take of your remark?

Yes. But was just a non-blocking comment, I could live with the ```if``` :)

**Pull request merged**